mbus/nats_handler.go

```diff
@@ -130,17 +130,35 @@ func (h *natsHandler) Start(handlerFunc boshhandler.Func) error {
 	h.logger.Info(h.logTag, "Attempting to connect to NATS")
 
 	if net.ParseIP(connectionInfo.IP) != nil {
-		err = h.platform.DeleteARPEntryWithIP(connectionInfo.IP)
-		if err != nil {
-			h.logger.Error(h.logTag, "Cleaning ip-mac address cache for: %s", connectionInfo.IP)
-		}
+		h.arpClean()
 	}
-
 	var natsOptions = []nats.Option{
-		nats.Timeout(natsConnectionTimeout),
+		nats.DisconnectHandler(func(c *nats.Conn) {
+			for c.IsReconnecting() {
+				h.arpClean()
+				h.logger.Debug(natsHandlerLogTag, fmt.Sprintf("Waiting to reconnect to nats.. Connected: %v", c.IsConnected()))
+				time.Sleep(time.Second)
+			}
+			h.logger.Debug(natsHandlerLogTag, "Reconnected to nats")
+		}),
+		nats.ClosedHandler(func(c *nats.Conn) {
+			h.logger.Debug(natsHandlerLogTag, fmt.Sprintf("Nats closed with %v\nConnected: %v\n Reconnecting: %v", c.LastError(), c.IsConnected(), c.IsReconnecting()))
+			for c.IsReconnecting() {
+				h.arpClean()
+				h.logger.Debug(natsHandlerLogTag, fmt.Sprintf("Waiting to reconnect to nats.. Connected: %v", c.IsConnected()))
+				time.Sleep(time.Second)
+			}
+			h.logger.Debug(natsHandlerLogTag, "Reconnected to nats")
+		}),
+		nats.ErrorHandler(func(c *nats.Conn, s *nats.Subscription, err error) {
+			h.logger.Debug(natsHandlerLogTag, err.Error())
+		}),
+		nats.CustomReconnectDelay(func(attempts int) time.Duration {
+			reconnectWait := time.Duration(time.Duration(math.Pow(natsMinRetryWait, float64(attempts))) * time.Second)
+			h.logger.Debug(natsHandlerLogTag, fmt.Sprintf("Increased reconnect to: %v", reconnectWait))
+			return reconnectWait
+		}),
 		nats.MaxReconnects(natsConnectionMaxRetries),
-		nats.ReconnectWait(h.connectRetryInterval),
-		nats.ReconnectJitter(h.maxConnectRetryInterval, h.maxConnectRetryInterval),
```

Comment:
We don't define any `ReconnectJitter` any more but will use the default one https://github.com/nats-io/nats.go/blob/1714729895dfe3f5aa0cfee3984b89bebcfce23c/nats.go#L57. Is this wanted?

Reply:
That was replaced by adding the exponential backoff. Looking through the old commit messages the Idea was to introduce some variation to lessen the load on the nats server in case of restart.